### ISSUE TITLE
1.24.1 reset custom navbar settings

### ISSUE THREAD

**Issue by kccuber-scratch** [ScratchAddons/ScratchAddons#4221]
### Description

An update to SA v1.24.1 reset my custom navigation bar settings.
### Repoduction steps

1. Make sure you have 1.24.0
2. Mess around with custom navbar settings
3. Check that your settings applied
4. Update to 1.24.1
5. Check your navbar
6. Your navbar should have been reset to a default `Create | Explore | Discuss | Ideas | About`

### Expected behaviour

The navbar settings should stay.
### Actual behaviour

The custom navbar disappeared.
### Screenshots

sry, I dont have screenshots
### Environment

- Browser:  Firefox 96.0 (I should update, however I dont want to)
- Scratch Addons version:  v1.24.0
- Operating system:  Windows 10

**Comment by Norbiros** [comment]
It happened in other different addon? Can you please download settings before and after? 
![Screenshot from 2022-02-06 17-23-52](https://user-images.githubusercontent.com/89837724/152690506-7de4f8ee-aa03-4967-8781-6be01f5cc5d7.png)

**Comment by Samq64** [comment]
I can't reproduce with the navbar asdon.

**Comment by WorldLanguages** [comment]
@kccuber-scratch I couldn't reproduce successfully.

Could you clarify, were you able to reproduce? Or did this happen to you after the stable version updated in the Firefox store? (v1.24.0 didn't reach Chrome users)

**Comment by WorldLanguages** [comment]
I find this bug to be impossible in normal scenarios (users that downloaded the extension from the store and never went back to an older version). 

This is the only migration code for the discuss-button addon. In other words, this is the only code that would only affect this addon's settings after an update. Of course there's code that could affect it after an extension restart with the version remaining the same, but that's not the bug that's being reported.
```js
 if (
          addonId === "discuss-button" &&
          addonsEnabled["discuss-button"] === true &&
          (settings.buttonName || settings.removeIdeasBtn)
        ) {
          // Transition v1.23.0 modes to v1.24.0 settings
          madeChangesToAddon = true;
          madeAnyChanges = true;

          let option = manifest.settings.find((option) => option.id === "items");
          settings.items = [...option.default];
          settings.items.splice(2, 0, {
            name: settings.buttonName,
            url: "/discuss",
          });
          if (settings.removeIdeasBtn) settings.items.splice(3, 1);

          delete settings.removeIdeasBtn;
          delete settings.buttonName;
        }
```

It's impossible to recover the `removeIdeasBtn` and `buttonName` settings, they're deleted in the migration. So it's impossible that this if case happens twice.

**Comment by WorldLanguages** [comment]
I'll close soon unless @kccuber-scratch can provide a more detailed reproduction.

**Comment by devxan** [comment]
This problem arose in my Scratch Addons extension too. I had no idea at the time why the Scratch navigation bar had changed back to its normal state, but I easily was able to bring the settings back to what I wanted. It might've been for some other reason?

**Comment by kccuber-scratch** [comment]
> @kccuber-scratch I couldn't reproduce successfully.
> 
> Could you clarify, were you able to reproduce? Or did this happen to you after the stable version updated in the Firefox store? (v1.24.0 didn't reach Chrome users)

sry, it was 1.24.0 in Firefox, and it auto updated to 1.24.1 which deleted that addon's settings for some reason. It's kind of impossible to reproduce this issue now because like everybody has 1.24.1 or 1.24.2

**Comment by apple502j** [comment]
Resolved in #4240 

TLDR: A refactor in dark-www by @mxmou included a code change that would crash while loading custom navbar. This change was made in 1.24.1.

**Comment by apple502j** [comment]
Whoops, this is about custom navbar, not forum toolbar...

If someone wants to repro this, please make sure to:
- 1) Clear internal storage in 1.24.0 first (`chrome.storage.sync.clear()`)
- 2) Preserve all logs, including background page logs and any logs on scratch.mit.edu website
- 3) Export settings as JSON both before and after version bump

@devxan @kccuber-scratch ?

**Comment by Weredime** [comment]
We can't modify previous versions
